libcaf_core/caf/flow/observable.hpp

```diff
@@ -861,7 +861,7 @@ struct has_max_concurrent_arg : std::false_type {};
 
 template <class T, class... Ts>
 struct has_max_concurrent_arg<T, Ts...> {
-  static constexpr bool value = std::is_unsigned_v<std::decay_t<T>>;
+  static constexpr bool value = std::is_same_v<std::remove_cvref_t<T>, size_t>;
```

Comment:
`remove_cvref_t` is C++20. `decay_t` would work.

Reply:
Ah, didn't think about that too deeply, it's become an instinct now to use that over decay. Will fix.